parachain/primitives/core/src/operating_mode.rs

```diff
@@ -0,0 +1,82 @@
+use codec::{Decode, Encode, FullCodec, MaxEncodedLen};
+use frame_support::{
+	dispatch::DispatchResult, traits::EnsureOrigin, PalletError, RuntimeDebug, StorageValue,
+};
+use scale_info::TypeInfo;
+use sp_std::fmt::Debug;
+
+/// Error generated by the `OwnedBridgeModule` trait.
+#[derive(Encode, Decode, TypeInfo, PalletError)]
+pub enum BridgeModuleError {
+	/// All pallet operations are halted.
+	Halted,
+}
+
+/// Operating mode for a bridge module.
+pub trait OperatingMode: Send + Copy + Debug + FullCodec {
+	// Returns true if the bridge module is halted.
+	fn is_halted(&self) -> bool;
+}
+
+/// Basic operating modes for a bridges module (Normal/Halted).
+#[derive(
+	Encode,
+	Decode,
+	Clone,
+	Copy,
+	PartialEq,
+	Eq,
+	RuntimeDebug,
+	TypeInfo,
+	MaxEncodedLen,
+	serde::Serialize,
+	serde::Deserialize,
+)]
+pub enum BasicOperatingMode {
+	/// Normal mode, when all operations are allowed.
+	Normal,
+	/// The pallet is halted. All operations (except operating mode change) are prohibited.
+	Halted,
+}
+
+impl Default for BasicOperatingMode {
+	fn default() -> Self {
+		Self::Normal
+	}
+}
+
+impl OperatingMode for BasicOperatingMode {
+	fn is_halted(&self) -> bool {
+		*self == BasicOperatingMode::Halted
+	}
+}
+
+/// Bridge module with operating mode
+pub trait BridgeModule<T: frame_system::Config> {
+	type OperatingMode: OperatingMode;
+	type OperatingModeStorage: StorageValue<Self::OperatingMode, Query = Self::OperatingMode>;
+	type AllowedHaltOrigin: EnsureOrigin<T::RuntimeOrigin>;
```

Comment:
This is some refactor for the [comment](https://github.com/Snowfork/snowbridge/pull/969#discussion_r1353662936) from @alistair-singh 

IMO though we remove `owner` and only allow root in this PR but with the `AllowedHaltOrigin` abstraction it's easy to extend to some other origins without require adding extrinsic like `set_owner`.

But anyway, let me know if you all think this is unnessessary and I'll revert that [commit](https://github.com/Snowfork/snowbridge/commit/aa4fb3d60d084e4cb603c169bdd8348f1e5d9c47).

Reply:
Reverted in
https://github.com/Snowfork/snowbridge/pull/969/commits/04057d72fad0434eaca1a0a93e787214ce17b509

@alistair-singh  Since the main target of this PR is to remove other origins and only allow `sudo`. So perhaps make things simple is better.

Reply:
I am happy with the revert.